packages/aws-cdk/lib/toolkit/cli-io-host.ts

```diff
@@ -133,6 +182,50 @@ export class CliIoHost {
   }
 }
 
+/**
+ * Validates a message code. The code must:
+ * - Be in the format [A-Z]+_[0-2][0-9]{3}
+ * - Have a numeric prefix that matches the message level:
+ *   - Error level codes must begin with 0
+ *   - Warning level codes must begin with 1
+ *   - Info/debug/trace level codes must begin with 2
+ *
+ * @param code The message code to validate
+ * @param level The IoMessageLevel of the message
+ * @returns boolean indicating if the code is valid for the given level
+ */
+export function validateMessageCode(code: string, level?: IoMessageLevel): boolean {
+  // Basic format validation
+  const pattern = /^[A-Z]+_([0-2])\d{3}$/;
+  const match = code.match(pattern);
+
+  if (!match) {
+    return false;
+  }
+
+  // If no level is provided, only validate the format
+  if (!level) {
+    return true;
+  }
+
+  // Extract the level number from the code
+  const levelNum = parseInt(match[1]);
+
+  // Validate level-specific prefix
+  switch (level) {
+    case 'error':
+      return levelNum === 0;
+    case 'warn':
+      return levelNum === 1;
+    case 'info':
+    case 'debug':
+    case 'trace':
+      return levelNum === 2;
+    default:
```

Comment:
should be no need for default if you get all the levels

Reply:
You never know what someone is gonna add and forget to modify in the validation function 😅